server/src/main/java/org/elasticsearch/action/admin/indices/forcemerge/TransportForceMergeAction.java

```diff
@@ -92,12 +93,16 @@ protected void shardOperation(
         ActionListener<TransportBroadcastByNodeAction.EmptyResult> listener
     ) {
         assert (task instanceof CancellableTask) == false; // TODO: add cancellation handling here once the task supports it
-        threadPool.executor(ThreadPool.Names.FORCE_MERGE).execute(ActionRunnable.supply(listener, () -> {
+        SubscribableListener.<IndexShard>newForked(l -> {
```

Comment:
Nit: seems to be neater to initialize indexShard and use `SubscribableListener<Void>` for  `indexShard.ensureMutable`?

```java
IndexShard indexShard = indicesService.indexServiceSafe(shardRouting.shardId().getIndex()).getShard(shardRouting.shardId().id());
SubscribableListener.<Void>newForked(l -> indexShard.ensureMutable(l))
    .<EmptyResult>andThen(l -> threadPool.executor(ThreadPool.Names.FORCE_MERGE).execute(ActionRunnable.supply(l, () -> {
        indexShard.forceMerge(request);
        return EmptyResult.INSTANCE;
    })))
    .addListener(listener);
```

Reply:
We need to do the error handling when the index doesn't exist in the node anymore, that's why I included it in the listener.